llvm/lib/Target/AIE/AIESlotCounts.h

```diff
@@ -0,0 +1,55 @@
+//===- AIESlotCounts.h - Resource computation utility
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+//
+//===----------------------------------------------------------------------===//
+// This defines a class that can be used to tally up the slots required for
+// one or more instructions
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_LIB_TARGET_AIE_AIESLOTCOUNTS_H
+#define LLVM_LIB_TARGET_AIE_AIESLOTCOUNTS_H
+
+#include "MCTargetDesc/AIEFormat.h"
+
+namespace llvm {
+namespace AIE {
+
+/// Efficient representation of slot requirements
+class SlotCounts {
+  static const int MaxSlots = 16;
+  short Counts[MaxSlots];
```

Comment:
Nit: given that `MaxSlots` is only `16`, I don't think we save so much memory going to `short`. Or do you think `SlotCounts` will be used extensively and the size difference will matter?